src/action/common/configure_shell_profile.rs

```diff
@@ -1,51 +1,142 @@
-use crate::action::base::CreateOrAppendFile;
+use crate::action::base::{CreateDirectory, CreateOrAppendFile};
 use crate::action::{Action, ActionDescription, ActionError, StatefulAction};
 
-use std::path::Path;
+use nix::unistd::User;
+use std::path::{Path, PathBuf};
 use tokio::task::JoinSet;
 
+// Fish has different syntax than zsh/bash, treat it separate
+const PROFILE_FISH_SUFFIX: &str = "conf.d/nix.fish";
+
+/**
+ Each of these are common values of $__fish_sysconf_dir,
+under which Fish will look for a file named
+[`PROFILE_FISH_SUFFIX`].
+*/
+const PROFILE_FISH_PREFIXES: &[&str] = &[
+    "/etc/fish",              // standard
+    "/usr/local/etc/fish",    // their installer .pkg for macOS
+    "/opt/homebrew/etc/fish", // homebrew
+    "/opt/local/etc/fish",    // macports
+];
 const PROFILE_TARGETS: &[&str] = &[
     "/etc/bashrc",
     "/etc/profile.d/nix.sh",
     "/etc/zshrc",
     "/etc/bash.bashrc",
     "/etc/zsh/zshrc",
-    // TODO(@hoverbear): FIsh
 ];
-const PROFILE_NIX_FILE: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh";
+const PROFILE_NIX_FILE_SHELL: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh";
+const PROFILE_NIX_FILE_FISH: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish";
 
 /**
 Configure any detected shell profiles to include Nix support
  */
 #[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
 pub struct ConfigureShellProfile {
+    create_directories: Vec<StatefulAction<CreateDirectory>>,
     create_or_append_files: Vec<StatefulAction<CreateOrAppendFile>>,
 }
 
 impl ConfigureShellProfile {
     #[tracing::instrument(skip_all)]
     pub async fn plan() -> Result<StatefulAction<Self>, ActionError> {
         let mut create_or_append_files = Vec::default();
+        let mut create_directories = Vec::default();
+
+        let shell_buf = format!(
+            "\n\
+            # Nix\n\
+            if [ -e '{PROFILE_NIX_FILE_SHELL}' ]; then\n\
+            {inde}. '{PROFILE_NIX_FILE_SHELL}'\n\
+            fi\n\
+            # End Nix\n
+        \n",
+            inde = "    ", // indent
```

Comment:
Is there a reason this isn't the full `indent`, or even just inlined...?

Reply:
I made it 4 characters so it was the same size as the 4 space ident, I made this to prevent the created content in the files have the ident.